statsmodels/tsa/statespace/mlemodel.py

```diff
@@ -0,0 +1,797 @@
+            instead.
+        full_results : boolean, optional
+            If True, returns a FilterResults instance; if False returns a
+            tuple with forecasts, the forecast errors, and the forecast error
+            covariance matrices. Default is False.
+
+        Returns
+        -------
+        forecast : array
+            Array of out of sample forecasts.
+        """
+        if start is None:
+            start = 0
+
+        # Handle start and end (e.g. dates)
+        start = self.model._get_predict_start(start)
+        end, out_of_sample = self.model._get_predict_end(end)
+
+        # Handle string dynamic
+        dates = self.data.dates
+        if isinstance(dynamic, str):
+            if dates is None:
+                raise ValueError("Got a string for dynamic and dates is None")
+            dtdynamic = self.model._str_to_date(dynamic)
+            try:
+                dynamic_start = self.model._get_dates_loc(dates, dtdynamic)
+
+                dynamic = dynamic_start - start
+            except KeyError:
+                raise ValueError("Dynamic must be in dates. Got %s | %s" %
+                                 (str(dynamic), str(dtdynamic)))
+
+        # Perform the prediction
+        results = super(MLEResults, self).predict(
+            start, end+out_of_sample+1, dynamic, full_results, *args, **kwargs
+        )
+
+        # Note: to be consistent with Statsmodels, return only the forecasts
+        # unless full_results is specified. Confidence intervals and the date
+        # indices are left out for now, but will likely be moved to a separate
+        # function in the future.
+        if full_results:
+            return results
+        else:
+            # (forecasts, forecasts_error, forecasts_error_cov) = results
+            forecasts = results
+
+        # Calculate the confidence intervals
+        # critical_value = norm.ppf(1 - alpha / 2.)
+        # std_errors = np.sqrt(forecasts_error_cov.diagonal().T)
+        # confidence_intervals = np.c_[
+        #     (forecasts - critical_value*std_errors)[:, :, None],
+        #     (forecasts + critical_value*std_errors)[:, :, None],
+        # ]
+
+        # Return the dates if we have them
+        # index = np.arange(start, end+out_of_sample+1)
+        # if hasattr(self.data, 'predict_dates'):
+        #     index = self.data.predict_dates
+        #     if(isinstance(index, pd.DatetimeIndex)):
+        #         index = index._mpl_repr()
+
+        return forecasts
+
+    def forecast(self, steps=1, *args, **kwargs):
+        """
+        Out-of-sample forecasts
+
+        Parameters
+        ----------
+        steps : int, optional
+            The number of out of sample forecasts from the end of the
+            sample. Default is 1.
+
+        Returns
+        -------
+        forecast : array
+            Array of out of sample forecasts.
+        """
+        return self.predict(start=self.nobs, end=self.nobs+steps-1,
```

Comment:
Positional parameters after keyword parameters.  Also, are the keyword parameters immutable (`self.nobs`)?  Probably by the looks to things.